Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -135,15 +135,23 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
         return;
     }
 
-    // Cap expiry of incoming cookies in response if it is a same-site
-    // subresource but it resolves to a different CNAME than the top
-    // site request, a.k.a. third-party CNAME cloaking.
+    // Cap expiry of incoming cookies in response if it is a same-site subresource but
+    // it resolves to a different CNAME or IP address range than the top site request,
+    // i.e. third-party CNAME or IP address cloaking.
     auto firstPartyURL = request.firstPartyForCookies();
     auto firstPartyHostName = firstPartyURL.host().toString();
-    auto firstPartyHostCNAME = m_networkSession->firstPartyHostCNAMEDomain(firstPartyHostName);
-    auto firstPartyAddress = m_networkSession->firstPartyHostIPAddress(firstPartyHostName);
 
-    task()._cookieTransformCallback = makeBlockPtr([requestURL = crossThreadCopy(request.url()), firstPartyURL = crossThreadCopy(firstPartyURL), firstPartyHostCNAME = crossThreadCopy(firstPartyHostCNAME), firstPartyAddress = crossThreadCopy(firstPartyAddress), thirdPartyCNAMEDomainForTesting = crossThreadCopy(m_networkSession->thirdPartyCNAMEDomainForTesting()), ageCapForCNAMECloakedCookies = crossThreadCopy(m_ageCapForCNAMECloakedCookies), weakTask = WeakObjCPtr<NSURLSessionTask>(task()), debugLoggingEnabled = m_networkSession->networkStorageSession()->trackingPreventionDebugLoggingEnabled()] (NSArray<NSHTTPCookie*> *cookiesSetInResponse) -> NSArray<NSHTTPCookie*> * {
+    task()._cookieTransformCallback = makeBlockPtr([
+        requestURL = crossThreadCopy(request.url())
+        , firstPartyURL = crossThreadCopy(firstPartyURL)
+        , firstPartyHostCNAME = crossThreadCopy(m_networkSession->firstPartyHostCNAMEDomain(firstPartyHostName))
+        , firstPartyAddress = crossThreadCopy(m_networkSession->firstPartyHostIPAddress(firstPartyHostName))
+        , thirdPartyCNAMEDomainForTesting = crossThreadCopy(m_networkSession->thirdPartyCNAMEDomainForTesting())
+        , ageCapForCNAMECloakedCookies = crossThreadCopy(m_ageCapForCNAMECloakedCookies)
+        , weakTask = WeakObjCPtr<NSURLSessionTask>(task())
+        , topDomainName = crossThreadCopy(RegistrableDomain { firstPartyURL }.string())
```

Comment:
Let's call it `topRegistrableDomain` instead. See comment further down.

Reply:
Or `firstPartyRegistrableDomain` to match the others.

Reply:
Good point! I'll go with `firstPartyRegistrableDomain`.